pkg/controller/machinepool/awsactuator.go

```diff
@@ -531,3 +692,54 @@ func (a *AWSActuator) validateSubnets(subnets map[string]ec2.Subnet, pool *hivev
 	}
 	return subnetsByAvailabilityZone, nil
 }
+
+func AWSMachineTemplate(infraName, ami string, nodePool *hivev1.MachinePool, subnetRef *capiaws.AWSResourceReference, controlPlaneNamespace, zone string) (*capiaws.AWSMachineTemplate, string) {
```

Comment:
some comment that explains that the string returns is a hash, and what we can use that hash for?